### PR TITLE
Handling some timeout on windows.

### PR DESCRIPTION
**Pull request by OCamlPro-Henry**
This is the minimal changes required to include https://reviews.facebook.net/D45849 in Hack.

### CODE REVIEW

File: hack/watchman/watchman.ml [hunk 1]
```diff
@@ -184,9 +184,10 @@ let get_changes env =
   set_of_list @@ extract_file_names env response
 
 let get_sockname timeout =
-  let ic = Unix.open_process_in "watchman get-sockname --no-pretty" in
+  let ic =
+    Timeout.open_process_in "watchman" [|"watchman"; "get-sockname"; "--no-pretty" |] in
```
Comment (gabelevi) [inline hack/watchman/watchman.ml]: This is a mistake. I'm fixing it though

**Comment by gabelevi** [comment]
This PR I think had a bad rebase and included some hh_matcher changes. I'm handling it though.

File: hack/server/serverMain.ml [hunk 2]
```diff
@@ -331,18 +329,19 @@ let run_load_script genv cmd =
...
       in
-      with_timeout
-        (ServerConfig.load_script_timeout genv.config)
+      Timeout.read_process
+        ~timeout:(ServerConfig.load_script_timeout genv.config)
         ~on_timeout:(fun _ -> failwith "Load script timed out")
-        ~do_:do_fn
-    in
+        ~reader
+        cmd [| cmd |] in
```
Comment (gabelevi) [inline hack/server/serverMain.ml]: This is also wrong. I'm fixing it though.

**Comment by gabelevi** [comment]
Ok, I have a clean version of this PR internally that is building for everyone and passing all the tests. I'll get someone else to double check my work and then we'll merge. Thanks for your patience!

**Comment by OCamlPro-Henry** [comment]
> This PR I think had a bad rebase and included some hh_matcher changes. I'm handling it though.

Thanks, and sorry for the puzzling mistakes.

**Comment by OCamlPro-Bozman** [comment]
Please check [#1183](https://github.com/facebook/flow/pull/1183).

**Pull request closed without merging**